### PR TITLE
Add executionContext/expressionValuesComputed notifincation

### LINKED ISSUE #589: Implement `executionContext/expressionValuesComputed`
### Summary
In order to do anything useful with `executionContext/push`, the IDE needs to be able to understand the changes that are made as a result of that operation. That's where this notification comes in.

### Value
The IDE will get notifications about changes to computed expressions as a result of its operations.

### Specification

- [ ] Implement `executionContext/expressionValuesComputed`, as specified in the [design docs](https://github.com/luna/enso/blob/master/doc/design/engine/engine-services.md#executioncontextexpressionvaluescomputed)
- [ ] Document any additional errors in the documentation.

### Acceptance Criteria & Test Cases

- The above specification has been implemented.
- The above functionality has been tested.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #589 

When the client creates execution context, `ContextEventsListener` starts listening runtime updates for this context. `ContextEventListener` is stopped when the client destroys the context.

Changelog:

- add: `ContextEventsListener` created per context, listens to runtime updates, and notifies the client
- refactor: add `client` argument to the request handlers

### Important Notes

When `ContextEventListener` fails to convert an update from the runtime, this update is ignored and the error is logged.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextRegistry.scala [hunk 1]
```diff
@@ -20,12 +20,20 @@ import org.enso.polyglot.runtime.Runtime.Api
   * Legend:
   *
   *   - 1  - Singleton
+  *   - *C - Created per client.
```
Comment (iamrecursion) [inline engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextRegistry.scala:23]: This kind of thing should be in the design documentation, not inline in the file. The doc comment should link to the design documentation.

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextRegistryProtocol.scala [hunk 2]
```diff
@@ -76,6 +76,17 @@ object ContextRegistryProtocol {
     */
   case class PopContextResponse(contextId: ContextId)
 
+  /**
+    * A notification that new information about some expressions is available.
+    *
+    * @param contextId execution context identifier
+    * @param updates a list of updated expressions
+    */
+  case class ExpressionValuesComputed(
```
Comment (iamrecursion) [inline engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextRegistryProtocol.scala:85]: This should probably be `ExpressionValuesComputedNotification` given that everything above is `FooBarRequest`. Alternatively remove `Request` from everything else.

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextEventsListener.scala [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        }
+      }
+      client.actor ! ContextRegistryProtocol
+        .ExpressionValuesComputedNotification(
+          contextId,
+          updates
+        )
+  }
+
+  private def getRuntimeUpdate(
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextEventsListener.scala:49]: It's not a getter, it is a converter.

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextEventsListener.scala [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            update.expressionId,
+            update.expressionType,
+            update.shortValue,
+            Some(pointer)
+          )
+        }
+    }
+  }
+
+  private def getRuntimePointer(
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextEventsListener.scala:74]: Please rename it to convert or create

File: engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextEventsListener.scala [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+  private def getRelativePath(path: java.nio.file.Path): Option[Path] =
+    config.contentRoots.view.flatMap {
+      case (id, root) =>
+        if (path.startsWith(root.toPath)) {
+          Some(Path(id, root.toPath.relativize(path)))
+        } else {
+          None
+        }
+    }.headOption
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/runtime/ContextEventsListener.scala:93]: Please move it the Config class as findRelativePath

**Pull request merged**